### PR TITLE
Move selected addresses for permissions to app db

### LINKED ISSUE #18405: Addresses for permissions screen: Move selected addresses state to app-db
Follow-up of #18382 

Relevant discussion,
- https://github.com/status-im/status-mobile/pull/18382#discussion_r1442689518
- https://github.com/status-im/status-mobile/pull/18382#discussion_r1442699882

### PR DESCRIPTION
**Pull request by ajayesivan**
fixes #18405 

status: ready

### CODE REVIEW

File: src/status_im/contexts/communities/events.cljs [hunk 1]
```diff
@@ -185,6 +185,37 @@
...
-(rf/reg-event-fx :communities/set-addresses-for-permissions
- (fn [{:keys [db]} [addresses]]
-   {:db (assoc-in db [:communities/addresses-for-permissions] addresses)}))
+(defn initialize-permission-addresses
+  [{:keys [db]}]
+  (let [accounts  (get-in db [:wallet :accounts])
+        addresses (set (map :address (vals accounts)))]
+    {:db (assoc db
+                :communities/previous-permission-addresses addresses
+                :communities/selected-permission-addresses addresses)}))
```
Comment (ajayesivan) [inline src/status_im/contexts/communities/events.cljs:194]: `:communities/selected-permission-addresses` - This key will track the selected address for both passing it to the status-go API and displaying it in the UI.

`:communities/previous-permission-addresses` - This key is specifically for storing the previously selected state. It enables us to reset to the previous address when the user clicks 'Cancel' after making an edit. 

@ilmotta, I hope these new key names make more sense.

Comment (ilmotta) [reply src/status_im/contexts/communities/events.cljs:194]: It's perfect @ajayesivan. You even managed to come up with names with the same length, god mode.

File: src/status_im/subs/root.cljs [hunk 2]
```diff
@@ -145,7 +145,8 @@
 (reg-root-key-sub :communities/collapsed-categories :communities/collapsed-categories)
 (reg-root-key-sub :communities/selected-tab :communities/selected-tab)
 (reg-root-key-sub :contract-communities :contract-communities)
-(reg-root-key-sub :communities/addresses-for-permissions :communities/addresses-for-permissions)
+(reg-root-key-sub :communities/selected-permission-addresses
```
Comment (ilmotta) [inline src/status_im/subs/root.cljs]: The args fit on the same line.

**Review by ilmotta — APPROVED** [review]
Nicely done!

File: src/status_im/contexts/communities/events.cljs (hunk 1 above)
Comment (ajayesivan) [reply src/status_im/contexts/communities/events.cljs:194]: It was a pain 😓 I renamed it 4 or 5 times and finally ended up with this.

**Review by yqrashawn — APPROVED** [review]

**Pull request merged**